homeassistant/components/mqtt/sensor.py

```diff
@@ -140,16 +153,49 @@ def message_received(msg):
             self._state = payload
             self.async_write_ha_state()
 
+        topics["state_topic"] = {
+            "topic": self._config[CONF_STATE_TOPIC],
+            "msg_callback": message_received,
+            "qos": self._config[CONF_QOS],
+        }
+
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def last_reset_message_received(msg):
+            """Handle new last_reset messages."""
+            payload = msg.payload
+
+            template = self._config.get(CONF_LAST_RESET_VALUE_TEMPLATE)
+            if template is not None:
+                variables = {"entity_id": self.entity_id}
+                payload = template.async_render_with_possible_json_value(
+                    payload,
+                    self._state,
+                    variables=variables,
+                )
+            if not payload:
+                _LOGGER.debug("Ignoring empty last_reset message from '%s'", msg.topic)
+                return
+            try:
+                last_reset = dt_util.parse_datetime(payload)
```

Comment:
We don't store datetime in the state machine, so there is no need to store this as a datetime in a property. I would just forward the data as-is. 

We should also make sure we document in the dev docs what the isoformat is for `datetime.fromtimestamp(0)`, such that MQTT devices can send that as-is.

Reply:
`last_reset` is documented as returning a `datetime` though: https://developers.home-assistant.io/docs/core/entity/sensor
We could update it to allow either a `datetime` or a n ISO-formatted string?

Reply:
How does the rest of MQTT work? Do they validate incoming data?

Reply:
Yes, we validate the data.
In this case we want to make sure the resulting timestamp in the state attribute is in UTC too, so better to parse it as a datetime.